From figo1802@gmail.com  Fri Jul 10 15:05:08 2009
From: "Figo.zhang" <figo1802@gmail.com>
Date: Sat, 06 Jun 2009 19:11:31 +0800
Subject: Staging: comedi: no need for checking vfree call
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: Ian Abbott <abbotti@mev.co.uk>, Frank Mori Hess <fmhess@users.sourceforge.net>, David Schleef <ds@schleef.org>
Message-ID: <1244286691.3185.42.camel@myhost>


vfree() does it's own NULL checking,so no need for check before
calling it.

'pages' is local variable argment,so in v2, it is no need assignment
 to NULL.

Signed-off-by: Figo.zhang <figo1802@gmail.com>
Acked-by: Pekka Enberg <penberg@cs.helsinki.fi>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/drivers/staging/comedi/drivers.c
+++ b/drivers/staging/comedi/drivers.c
@@ -509,9 +509,8 @@ int comedi_buf_alloc(struct comedi_devic
 			async->prealloc_buf =
 			    vmap(pages, n_pages, VM_MAP, PAGE_KERNEL_NOCACHE);
 		}
-		if (pages) {
-			vfree(pages);
-		}
+		vfree(pages);
+
 		if (async->prealloc_buf == NULL) {
 			/* Some allocation failed above. */
 			if (async->buf_page_list) {
